hypothesis-python/src/hypothesis/core.py

```diff
@@ -735,6 +735,67 @@ def execute(data, function):
     return default_executor
 
 
+@contextlib.contextmanager
+def unwrap_exception_group() -> Generator[None, None, None]:
```

Comment:
We might want a name that indicates _what_ we're unwrapping, since this might be confused for a generic unwrap-single-exception helper (which is usually a bad idea, but...).  Maybe `unwrap_markers_from_group`?